pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -2275,9 +2296,18 @@ func (cc *XController) Cleanup(appwrapper *arbv1.AppWrapper) error {
 		// klog.Infof("[Dispatcher] Cleanup: State=%s\n", appwrapper.Status.State)
 		//if ! appwrapper.Status.CanRun && appwrapper.Status.IsDispatched {
 		if appwrapper.Status.IsDispatched {
-			queuejobKey, _ := GetQueueJobKey(appwrapper)
-			if obj, ok := cc.dispatchMap[queuejobKey]; ok {
-				cc.agentMap[obj].DeleteJob(appwrapper)
+
+			if cc.serverOption.ExternalDispatch {
+				if err := cc.arbclients.ArbV1().AppWrappers(appwrapper.Namespace).Delete(appwrapper.Name, &metav1.DeleteOptions{}); err != nil {
```

Comment:
A delete is already issued when code reaches this point.  There is no need to delete the AW again.  Probably only need to remove it from the book-keeping memory objects.